runner/models/guidelines.py

```diff
@@ -180,6 +180,28 @@ def __init__(self, name: str, children: list):
                 Guideline("Always import the schema definition functions from `convex/server`:"),
             ],
         ),
+        GuidelineSection(
+            "typescript_guidelines",
+            [               
+                Guideline(
+                    "You can use the helper typescript type 'Id' from the _generated/dataModel import to get the type of the id for a given table. For example if there is a table called 'users' you can use Id<'users'> to get the type of the id for that table."
+                ),
+                Guideline(
+                    "If you need to define a Record make sure that you correctly provide the type of the key and value in the type. For example a validator `v.record(v.id('users'), v.string())` would have the type `Record<Id<'users'>, string>`"
+                ),      
+                Guideline(
+                    "You should always prefer to be strict with the types particularly around id's of documents. So for example instead of a function taking in a string as an argument you should prefer to take in an Id<T> where T is the name of the table."
```

Comment:
```suggestion
                    "Be strict with types, particularly around id's of documents. For example, if a function takes in an id for a document in the 'users' table, take in `Id<'users'>` rather than `string`.
```
a little more concise and concrete. I think the models are generally better at generalizing from a concrete example rather than applying an abstract concept (like the generic with `Id<T>`).